homeassistant/components/sensor/hue.py

```diff
@@ -0,0 +1,513 @@
+        super().__init__(sensor, request_bridge_update, bridge)
+
+    @property
+    def state(self):
+        """Return the state of the Hue sensor."""
+        return pow(10, (self.sensor.lightlevel - 1) / 10000)
+
+    @property
+    def device_class(self):
+        """Return the device class of the Hue sensor."""
+        return DEVICE_CLASS_ILLUMINANCE
+
+    @property
+    def unit_of_measurement(self):
+        """Return the uom of the Hue sensor."""
+        return UOM_ILLUMINANCE
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = {}
+        attributes[ATTR_BATTERY_LEVEL] = self.sensor.battery
+        attributes[ATTR_DARK] = self.sensor.dark
+        attributes[ATTR_DAYLIGHT] = self.sensor.daylight
+        attributes[ATTR_LAST_UPDATED] = self.sensor.lastupdated
+        return attributes
+
+
+class ZLLSwitch(ZLLSensor):
+    """Class representing a ZLL (Hue Wireless Dimmer) Switch."""
+    def __init__(self, sensor, request_bridge_update, bridge):
+        super().__init__(sensor, request_bridge_update, bridge)
+        self._last_update_time = self.sensor.lastupdated
+
+    @property
+    def state(self):
+        """Return the switch's buttonevent if the lastupdated variable changed.
+        If lastupdated hasn't changed, reset the sensor's state to None."""
+        current_time = self.sensor.lastupdated
+        if current_time == self._last_update_time:
+            return None
+        else:
+            self._last_update_time = current_time
+            if self.sensor.buttonevent == ZLL_SWITCH_BUTTON_1_SHORT_RELEASED:
+                return '1_click'
+            elif self.sensor.buttonevent == ZLL_SWITCH_BUTTON_1_LONG_RELEASED:
+                return '1_hold'
+            elif self.sensor.buttonevent == ZLL_SWITCH_BUTTON_2_SHORT_RELEASED:
+                return '2_click'
+            elif self.sensor.buttonevent == ZLL_SWITCH_BUTTON_2_LONG_RELEASED:
+                return '2_hold'
+            elif self.sensor.buttonevent == ZLL_SWITCH_BUTTON_3_SHORT_RELEASED:
+                return '3_click'
+            elif self.sensor.buttonevent == ZLL_SWITCH_BUTTON_3_LONG_RELEASED:
+                return '3_hold'
+            elif self.sensor.buttonevent == ZLL_SWITCH_BUTTON_4_SHORT_RELEASED:
+                return '4_click'
+            elif self.sensor.buttonevent == ZLL_SWITCH_BUTTON_4_LONG_RELEASED:
+                return '4_hold'
+
+    @property
+    def icon(self):
+        """Return the remote icon."""
+        return ICON_REMOTE
+
+
+class ZLLTemperatureSensor(ZLLSensor):
+    """Representation of a Hue ZLL Temperature sensor."""
+
+    def __init__(self, sensor, request_bridge_update, bridge):
+        """Initialize the sensor."""
+        super().__init__(sensor, request_bridge_update, bridge)
+
+    @property
+    def state(self):
+        """Return the state of the Hue sensor."""
+        return round(self.sensor.temperature / 100, 1)
```

Comment:
Is there any difference between this temp sensor and the clip temp sensor? There's so much duplication of code going on here.